resources/public/js/index.js

```diff
@@ -1,11 +1,57 @@
 window.onload = () => {
-    const purge_dialog = document.querySelector(".purge-dialog");
-    const showDialog = document.querySelector(".purge-dialog-show")
-    const cancelDialog = document.querySelector("dialog .cancel")
-    showDialog.addEventListener("click", () => {
-        purge_dialog.showModal();
+  const purge_dialog = document.querySelector(".purge-dialog");
+  const showDialog = document.querySelector(".purge-dialog-show")
+  const cancelDialog = document.querySelector("dialog .cancel")
+  if (purge_dialog) {
+    showDialog.addEventListener("click", (event) => {
+      purge_dialog.showModal()
     })
-    cancelDialog.addEventListener("click", () => {
-        purge_dialog.close();
+    cancelDialog.addEventListener("click", (event) => {
+      purge_dialog.close()
     })
+  }
+
+  function createFilterValueInputElement() {
+    const input = document.createElement("input");
+    input.setAttribute('type', 'text');
+    input.setAttribute('name', 'filter-value');
+    input.setAttribute('class', 'filter-value');
+    input.setAttribute('placeholder', 'filter value')
+    input.setAttribute('value', '');
+    return input;
+  }
+  function createFilterValueSelectElement() {
+    const select = document.createElement('select');
+    select.setAttribute('id', 'job-type-select')
+    select.setAttribute('name', 'filter-value')
+    select.setAttribute('class', 'filter-value')
+    const options = ["unexecuted", "failed"];
+    options.forEach(function (t) {
+      var option = document.createElement('option');
+      option.value = t;
+      option.textContent = t;
+      select.appendChild(option);
+    });
+    return select;
+  }
+
+  const filterType = document.querySelector(".filter-type")
+
+  filterType.addEventListener("change", (event) => {
+    const parentDiv = document.querySelector(".filter-values")
+    const filterTypeValue = filterType.value
+    const filterValue = document.querySelector(".filter-opts-items .filter-value")
```

Comment:
suggestion: `filterType`, `filterTypeValue` and `filterValue` are a bit confusing. Can we consider renaming them to something more clearer or leave comments explaining them?